imap_processing/ena_maps/utils/corrections.py

```diff
@@ -321,72 +321,69 @@ class or child classes.
         return corrected_flux, corrected_flux_stat_unc
 
 
-def _add_spacecraft_velocity_to_pset(
-    pset: LoHiBasePsetSubclass,
-) -> LoHiBasePsetSubclass:
+def add_spacecraft_velocity_to_pset(
+    pset: xr.Dataset,
+) -> xr.Dataset:
     """
-    Calculate and add spacecraft velocity data to pointing set.
+    Calculate and add spacecraft velocity data to pointing set dataset.
 
     Parameters
     ----------
-    pset : LoHiBasePointingSet
-        Pointing set object to be updated.
+    pset : xr.Dataset
+        Pointing set dataset to be updated. Must contain "epoch" coordinate
+        and "epoch_delta" data variable.
 
     Returns
     -------
-    pset : LoHiBasePointingSet
-        Pointing set object with spacecraft velocity data added.
+    pset_processed : xarray.Dataset
+        Pointing set dataset with spacecraft velocity data added.
 
     Notes
     -----
-    Adds the following DataArrays to pset.data:
+    Adds the following DataArrays to input dataset:
     - "sc_velocity": Spacecraft velocity vector (km/s) with dims ["x_y_z"]
     - "sc_direction_vector": Spacecraft velocity unit vector with dims ["x_y_z"]
     """
     # Compute ephemeris time (J2000 seconds) of PSET midpoint time
-    # TODO: Use the Pointing midpoint time. Epoch should be start time
-    #     but use it until we can make Lo and Hi PSETs have a consistent
-    #     variable to hold the midpoint time.
-    et = ttj2000ns_to_et(pset.data["epoch"].values[0])
+    et = ttj2000ns_to_et(pset["epoch"].values[0] + pset["epoch_delta"].values[0] / 2)
```

Comment:
Might be useful to remind ourselves here with a comment that pset["epoch"] is the left edge and the delta is the full width.